src/file_compat.h

```diff
@@ -126,6 +126,9 @@ static NPY_INLINE FILE *mpl_PyFile_Dup(PyObject *file, char *mode, mpl_off_t *or
  */
 static NPY_INLINE int mpl_PyFile_DupClose(PyObject *file, FILE *handle, mpl_off_t orig_pos)
 {
+    PyObject *type = NULL, *value = NULL, *tb = NULL;
```

Comment:
These should probably get some differentiating name, like an `orig_` prefix or something.

Reply:
grepping through https://github.com/python/cpython/blob/f320be77ffb73e3b9e7fc98c37b8df3975d84b40/Modules/_io/fileio.c shows that cpython tends to stash the current exception (if any) in *exc, *val, *tb (but not without a prefix).  We're not shadowing anything anyways...

Reply:
Sure, but those are generally around one other call, whereas this one is around almost an entire function body.

Reply:
added exc_ prefix.